File: Frontend/src/interfaces/strategy-overview.ts
import { Indicator } from "./entities/indicator";
import { Strategy } from "./entities/strategy";
import type { Ticker } from "./ticker";
import { Pair } from "./entities/pair";
import { Asset } from "./entities/asset";
import { Status } from "./enums/status";
import { Action } from "./entities/action";

export interface StrategyOverview {
    strategy: Strategy;
    indicator: Indicator;
    action: Action;
    pair: Pair;
    base_asset: Asset;
    quote_asset: Asset;
    ticker: Ticker;
    order_status: Record<Status, number>;
}

File: Frontend/src-tauri/src/models/structs/strategy_overview.rs
use std::collections::HashMap;

use serde::{Deserialize, Serialize};

use crate::models::{
    entities::{actions, assets, indicators, pairs, strategies},
    enums::Status,
    structs::Ticker,
};

#[derive(Debug, Serialize, Deserialize, Default, Clone)]
pub struct StrategyOverview {
    pub strategy: strategies::Model,
    pub indicator: indicators::Model,
    pub action: actions::Model,
    pub pair: pairs::Model,
    pub base_asset: assets::Model,
    pub quote_asset: assets::Model,
    pub ticker: Ticker,
    pub order_status: HashMap<Status, i32>,
}
